poetry/console/commands/env/remove.py

```diff
@@ -1,21 +1,36 @@
 from cleo import argument
+from cleo import option
 
 from ..command import Command
 
 
 class EnvRemoveCommand(Command):
 
     name = "remove"
-    description = "Removes a specific virtualenv associated with the project."
+    description = "Removes specific virtualenvs associated with the project."
 
     arguments = [
-        argument("python", "The python executable to remove the virtualenv for.")
+        argument(
+            "python",
+            "The python executables to remove the virtualenv for.",
+            optional=True,
+            multiple=True,
+        )
+    ]
+    options = [
+        option("all", description="Remove all virtualenv associated with the project."),
     ]
 
     def handle(self):
         from poetry.utils.env import EnvManager
 
         manager = EnvManager(self.poetry)
-        venv = manager.remove(self.argument("python"))
-
-        self.line("Deleted virtualenv: <comment>{}</comment>".format(venv.path))
+        pythons = self.argument("python")
+        all = self.option("all")
+        if all:
```

Comment:
Might want to handle ambiguity condition where `all` and `python` is specified.

Reply:
Good point. I reordered the function so that it removes the python versions from the python argument and then if --all is provided then it removes any remaining environments. I think that will make the behavior more clear in the combined case. There is also a new test to validate this behavior.